gui/backend/upsert_bot_config_test.go

```diff
@@ -0,0 +1,70 @@
+package backend
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stellar/kelp/gui/model2"
+	"github.com/stellar/kelp/support/kelpos"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsBotNameValid(t *testing.T) {
+	testCases := []struct {
+		name      string
+		botName   string
+		strategy  string
+		wantValid bool
+	}{
+		{
+			name:      "success default",
+			botName:   "George the Friendly Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "success numeric",
+			botName:   "George the 1337 Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "success dash",
+			botName:   "George-the-Friendly-Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "failure <>",
+			botName:   "George<>the Friendly Octopus",
+			wantValid: false,
+		},
+		{
+			name:      "failure \\/",
+			botName:   "George\\/the Friendly Octopus",
+			wantValid: false,
+		},
+	}
+
+	for _, k := range testCases {
+		t.Run(k.name, func(t *testing.T) {
+			// errors would be thrown by the `regex` package, not due to kelp logic
+			gotValid, gotErr := isBotNameValid(k.botName)
+			assert.NoError(t, gotErr)
```

Comment:
if this is false then we should return so we don't produce false negatives for the remainder of the test:
```
if !assert.NoError(t, gotErr) {
    return
}
```

Reply:
can also use `e` instead of `gotErr` for errors that are not under test, which is a standardization across the codebase for tests.

using got in other places such as values (gotValid below) or errors being tested is ok.

Reply:
Makes sense, and `gotErr` use has been removed, since `isBotNameValid` no longer returns an error.